controllers/topology/internal/mergepatch/mergepatch.go

```diff
@@ -68,6 +68,17 @@ func NewHelper(original, modified client.Object, c client.Client, opts ...Helper
 		return nil, errors.Wrap(err, "failed to marshal modified object to json")
 	}
 
+	// Compute the merge patch that will align the original object to the target
+	// state defined above; this patch overrides the two-way merge patch for the
+	// authoritative paths, if any.
+	var authoritativePatch []byte
+	if helperOptions.authoritativePaths != nil {
+		authoritativePatch, err = jsonpatch.CreateMergePatch(originalJSON, modifiedJSON)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create merge patch")
```

Comment:
```suggestion
			return nil, errors.Wrap(err, "failed to create merge patch for authoritative paths")
```
nit: Maybe make the error message unique (vs l.93)